src/odemis/acq/stream/_live.py

```diff
@@ -430,12 +430,12 @@ def _applyROI(self):
         """
         res, shift = self._computeROISettings(self.roi.value)
 
-        if (self._shiftebeam == MTD_EBEAM_SHIFT) and (self._beamshift is not None):
-            shift = tuple(s + c for s, c in zip(shift, self._beamshift))
-
-        # always in this order
+        # always in this order: resolution, then shift
         self._emitter.resolution.value = res
-        self._emitter.shift.value = shift
+        if self._shiftebeam == MTD_EBEAM_SHIFT:
+            if self._beamshift is not None:
```

Comment:
What happens if self._beamshift == None? Shouldn't there be an error defined?

Reply:
No, beamshift == None is not sign of an error. It just means that there is no beamshift (known). That said, this could be as simply convey by setting beamshift to `(0, 0)`.
=> Also in my clean-up commit.